views/admin/product/all_products.php

```diff
@@ -8,40 +8,38 @@
 <div class='container'>
 <div class='row'>
 <div class='col-sm-12'>
-<?php
+<?php 
       try{
       echo "<br>";
       echo "<h1 class=text-center>All Products</h1>";
       echo "<br>";
         $stmt=$conn->prepare("SELECT * FROM `product`");
         $stmt->execute();
-        $result=$stmt->fetchAll();       
-         echo 
-      "<table class='table  table-hover'>
-         <thead>
+        $products=$stmt->fetchAll(); 
+        if(count($products)>0){     
+         echo "<table class='table  table-hover'>
+         <thead class='text-light bg-dark'>
            <tr>
              <th scope=col>Product_ID</th>
-             <th scope=col>Category_ID</th>
              <th scope=col>Name</th>
              <th scope=col>Price</th>
+             <th scope=col>Category_ID</th>
              <th scope=col>Product_Image</th>
-             <th scope=col>Product_Avilability</th>
              <th scope=col>Actions</th>
            </tr>
          </thead>";
-     foreach($result as $resul){
-     echo "<tbody> <tr> <td> ".$resul['product_id']. "</td>".
-      "<td> ".$resul['category_id']. "</td>".
-      "<td> ".$resul['name']. "</td>".
-      "<td> ".$resul['price']. "</td>".
-     "<td>"."<img src=".'../../../images/products/'.$resul['image']." width='150' height='100' style='border-radius:50%;'>"."</td>".
-     "<td> ".$resul['available']. "</td>".
-     "<td>" ."<button class='btn btn-danger'>Delete</button>
-     <button class='btn btn-primary'>Edit</button>" ."</td>".
-     "</tr> </tbody>";
-     }
-     echo "</table>";
-             
+          foreach($products as $product){
+            echo "<tr id='".$product['product_id']."'><th scope='row'>".$product['product_id']."</th>".              
+            "<td> ".$product['name']."</td>".
+            "<td> ".$product['price']."</td>".
+            "<td> ".$product['category_id']."</td>".
```

Comment:
You must not show data that was entered by the user to the screen before sanitizing it.

Reply:
Thanks for Your Notes , But Viewing Data is Assigned to Ahmed El Hadary Not To me.